### ISSUE TITLE
Core1 panic'ed StoreProhibited

### ISSUE THREAD

**Issue by gjongenelen** [whitecatboard/Lua-RTOS-ESP32#204]
I'm getting random panics when reading data via UART.

![screenshot 2018-10-25 at 17 00 10](https://user-images.githubusercontent.com/5789343/47509971-732f7d00-d877-11e8-8998-c9fc2a3998a6.png)

How can I debug this?

**Comment by thunderWhitecat** [comment]
@gjongenelen,

Please, proceed as follows:

make clean
make upgrade-idf
make

**Comment by gjongenelen** [comment]
Did that several times, but without any success.
I found out that it is the *el/*l-argument which is causing the issue.

```
uart.read(uart.UART1, "*el", 1000); -- StoreProhibited
uart.read(uart.UART1, "*l", 1000); -- StoreProhibited
uart.read(uart.UART1, "*c", 1000); -- No problems (so far)
```
I guess it has something to do with the buffer(?) in:
https://github.com/whitecatboard/Lua-RTOS-ESP32/blob/449517900a2ee4167628fc4f8547ce240e9de23d/components/lua/modules/hw/uart.c#L275

Important notice: The serial connection is mostly quiet. So it may happen that there are 30 timeouts before something enters the RX-line. (This is easy to reproduced by grabbing an ESP32, attaching UART without anything physically connected and reading constantly with timeout of 1000ms ex.)

**Comment by the0ne** [comment]
@gjongenelen you should make sure that `str` is not null after malloc, before using it in line 278.

https://github.com/whitecatboard/Lua-RTOS-ESP32/blob/449517900a2ee4167628fc4f8547ce240e9de23d/components/lua/modules/hw/uart.c#L275-L278

**Comment by the0ne** [comment]
fixed in 9e61e52725c4b77574e9bf9261a96f918ab779f5
as it currently only is in [my fork](https://github.com/the0ne/Lua-RTOS-ESP32) and the PR #184 you may want to cherry-pick it.

**Comment by thunderWhitecat** [comment]
@gjongenelen, @the0ne,

Merged in https://github.com/whitecatboard/Lua-RTOS-ESP32/commit/2b98d995192cc448af0499957635c5abbe8e4dea.
